csv/src/fs2/data/csv/ParseableHeader.scala

```diff
@@ -15,12 +15,16 @@
  */
 package fs2.data.csv
 
-/** A typeclass describing what it means to be a parseable
+import cats.data.NonEmptyList
+import cats.implicits._
+import cats.{Functor, SemigroupK}
+
+/** A type class describing what it means to be a parseable
   * CSV header.
   */
 trait ParseableHeader[Header] {
 
-  def parse(name: String): Header
+  def apply(names: NonEmptyList[String]): HeaderResult[Header]
```

Comment:
Why not use the `F` of the stream it will be used with?

Reply:
For representing errors? Consistency with `CsvRowDecoder`, `RowDecoder` and `CellDecoder` and more narrow error type. 

If we were to go with something like `def apply[F[_]](names: NonEmptyList[String])(implicit me: MonadError[F, Throwable]): F[NonEmptyList[Header]]`, other exception types could be thrown, if we use a `MonadError[F, HeaderError]`, it gets harder to use as there are usually no implicit instances available for that. Similar for `ApplicativeError`, but then you can't even `flatMap`. Might still be an interesting encoding if all decoder types were using it and some inductive instances for the necessary `MonadError` were defined in the project.

Making the effect type a parameter of `ParseableHeader` would allow for launching an array of missiles concurrently, I don't want header parsing to do that.

Reply:
I'm definitely not entirely happy with what I created here :( Ok, then.